compose/src/main/java/com/bottlerocketstudios/compose/snippets/CreateSnippet.kt

```diff
@@ -19,4 +60,168 @@ data class CreateSnippetScreenState(
 
 @Composable
 fun CreateSnippetScreen(state: CreateSnippetScreenState) {
+    val uiState by remember { mutableStateOf(state) }
+
+    Column(
+        modifier = Modifier
+            .fillMaxSize()
+    ) {
+        TextInputBox(
+            text = uiState.title.value,
+            onChanged = uiState.onTitleChanged,
+            hint = stringResource(id = R.string.create_snippet_title_hint),
+            topPadding = Dimens.grid_2
+        )
+        TextInputBox(
+            text = uiState.filename.value,
+            onChanged = uiState.onFilenameChanged,
+            hint = stringResource(id = R.string.create_snippet_filename_hint),
+        )
+        TextInputBox(
+            text = uiState.contents.value,
+            onChanged = uiState.onContentsChanged,
+            hint = stringResource(id = R.string.create_snippet_contents_hint),
+            minLines = 5,
+            imeAction = ImeAction.Done
+        )
+        LabelledCheckbox(uiState = uiState)
+        FailedText(uiState = uiState)
+        CreateSnippetButton(uiState = uiState)
```

Comment:
@VanyaHuamanBottleRocket @BottleRocket-Colin How do we feel about passing the whole uiState instance to each UI component onscreen?

1. Is it okay to pass the full thing? This is less ideal if needing to create a preview for the composable as the data class will need to be constructed with values that aren't used by the previewed composable.
2a. Should we try to always pass only the necessary values per component? b. If this gets fairly lengthy, should we create additional sub ui state data classes for specific composables that are subsets of the full screen ui state?
3. Should we have some wiggle room between 1 and 2? For example, if 7 of the 8 datapoints of ui state instance are needed for a composable then just pass the whole ui state object?

Initially, I'm kind of leaning towards 2a with a fallback to 3 when needed. How do you all feel about 2b vs 3?

Thoughts?

Reply:
I am ok with 2a. Let me know if I should update.

Reply:
@br-jared-stockton / @VanyaHuamanBottleRocket - Yes, I agree.  Let's break the state down and just send separate parameters for the particular function.  That will make things more re-useable.  On that point, I also agree that if 70-80% is used, there would be a large amount of parameters, and the function in mind is single use for this screen.  Then perhaps, pass the state thru, or create a  "sub" state for the particular widget, as Jared suggested above.

But we should def lean towards our nested composables using parameters and not the state model that was already consumed above it in heirarchy.

Reply:
@VanyaHuamanBottleRocket Let's move forward with 2a here, please sir!
@BottleRocket-Colin Agreed with your thoughts as well. Thanks for the input :)

Reply:
Updating.